xonsh/prompt/gitstatus.py

```diff
@@ -63,30 +42,32 @@ def _check_output(*args, **kwargs):
             raise
 
 
-@xl.lazyobject
-def _DEFS():
-    DEFS = {
-        "HASH": ":",
-        "BRANCH": "{CYAN}",
-        "OPERATION": "{CYAN}",
-        "STAGED": "{RED}●",
-        "CONFLICTS": "{RED}×",
-        "CHANGED": "{BLUE}+",
-        "DELETED": "{RED}-",
-        "UNTRACKED": "…",
-        "STASHED": "⚑",
-        "CLEAN": "{BOLD_GREEN}✓",
-        "AHEAD": "↑·",
-        "BEHIND": "↓·",
-        "LINES_ADDED": "{BLUE}+",
-        "LINES_REMOVED": "{RED}-",
-    }
-    return DEFS
+_DEFS = {
+    "HASH": ":",
+    "BRANCH": "{CYAN}",
+    "OPERATION": "{CYAN}",
+    "STAGED": "{RED}●",
+    "CONFLICTS": "{RED}×",
+    "CHANGED": "{BLUE}+",
+    "DELETED": "{RED}-",
+    "UNTRACKED": "…",
+    "STASHED": "⚑",
+    "CLEAN": "{BOLD_GREEN}✓",
+    "AHEAD": "↑·",
+    "BEHIND": "↓·",
+    "LINES_ADDED": "{BLUE}+",
+    "LINES_REMOVED": "{RED}-",
+}
+
+
+def _get_def(key: str) -> str:
+    def_ = XSH.env.get("XONSH_GITSTATUS_" + key.upper())
+    return def_ if def_ is not None else _DEFS[key]
```

Comment:
`key.upper()` is needed here too (put is in a variable and use it in both places)